src/kakarot/gas.cairo

```diff
@@ -82,4 +83,22 @@ namespace Gas {
 
         return new_cost - prev_cost;
     }
+
+    // @notive A proxy function for memory_expansion_cost that matches the actual computation
+    //         when offset and size are within range_check and returns a factor of the given
+    //         oog_cost otherwise, so as to trigger an OOG in any case but to avoid the more expensive
+    //         Uint256 based computing.
+    // @param words_len The current length of the memory as Uint256.
+    // @param max_offset The target max_offset to be applied to the given memory as Uint256.
+    // @param oog_cost The base cost to be returned in case of OOG. Should be enough to trigger OOG (eg. evm.gas_left).
+    // @return cost The current expansion gas cost.
+    func memory_expansion_cost_proxy{range_check_ptr}(
```

Comment:
I'd suggest
    `func memory_expansion_cost_with_oog{range_check_ptr}()`
Or
     `func bounded_memory_expansion_cost{range_check_ptr}()`
     `func memory_expansion_cost_bounded{range_check_ptr}()`


To only use the word `proxy` in our handling of account proxies

Reply:
I thought about `_bounded` or `_saturated` but it's not in the current implementation. But I can make it such that it's true, like return const value as if max_offset was 2**64 or 2**128